File: dev-docs/snippets/users/get-profiles.swift
import JsBaoClient

// Batch-fetch several basic profiles in one round-trip.
func getProfiles(client: JsBaoClient, ids: [String]) async throws {
  // #region example
  let profiles: [BatchUserProfile] = try await client.users.getProfiles(userIds: ids)
  let names = profiles.map { $0.name }
  // #endregion example
  _ = names
}

File: dev-docs/snippets/users/get-profiles.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Batch-fetch several basic profiles in one round-trip.
export async function getProfiles(client: JsBaoClient, ids: string[]) {
  // #region example
  const profiles = await client.users.getProfiles(ids);
  // #endregion example
  return profiles;
}
